### PR TITLE
Support profile channel

### PR DESCRIPTION
**Pull request by mrproliu**
* Using none queue mode to support the SkyWalking native profile channel

### CODE REVIEW

File: configs/satellite_config.yaml [hunk 1]
```diff
@@ -140,4 +140,28 @@ pipes:
...
\ No newline at end of file
+        - plugin_name: nativetracing-grpc-forwarder
+  - common_config:
+      pipe_name: profilepipe
+    gatherer:
+      server_name: "grpc-server"
+      receiver:
+        plugin_name: "grpc-nativeprofile-receiver"
+      queue:
+        plugin_name: "memory-queue"
```
Comment (wu-sheng) [inline configs/satellite_config.yaml:151]: This doesn't look like `none queue mode`. Could you be clear your meaning?

Comment (mrproliu) [reply configs/satellite_config.yaml:151]: in the profiling protocol, we also need to use a queue to transform thread dumps.

Comment (wu-sheng) [reply configs/satellite_config.yaml:151]: > Using none queue mode to support the SkyWalking native profile channel

What does this comment mean?

**Comment by wu-sheng** [comment]
This is only a profile snapshot reporting channel, right?

**Comment by mrproliu** [comment]
> This is only a profile snapshot reporting channel, right?

Also support the profile query and report finish channel, the profile channel mixes the none queue and queue data.

File: plugins/receiver/grpc/nativetracing/receiver.go [hunk 2]
```diff
@@ -52,6 +53,9 @@ func (r *Receiver) RegisterHandler(server interface{}) {
 	v3.RegisterTraceSegmentReportServiceServer(r.Server, r.service)
 }
 
+func (r *Receiver) RegisterSyncProcessor(_ api.SyncProcessor) {
```
Comment (wu-sheng) [inline plugins/receiver/grpc/nativetracing/receiver.go:56]: This is API definition, right? It seems missing comments.

**Review by wu-sheng — APPROVED** [review]
@EvanLjp Please recheck.

**Comment by EvanLjp** [comment]
OK，I would do review tomorrow, I just returned to Beijing today.

File: plugins/receiver/grpc/nativetracing/receiver.go (hunk 2 above)
Comment (mrproliu) [reply plugins/receiver/grpc/nativetracing/receiver.go:56]: It has been add comments on the `Receiver` interface. https://github.com/apache/skywalking-satellite/pull/53/files#diff-d7d2624abe5a66c8d5e294b65af7c537db3cece721746cec346e4582614d128eR37-R38

**Comment by EvanLjp** [comment]
How about adding a new module called invoker to do the sync process rather than coupling with the current gatherer, processor, and sender?

**Comment by mrproliu** [comment]
> How about adding a new module called invoker to do the sync process rather than coupling with the current gatherer, processor, and sender?

Thank you for the response. Would you do more instruction?

File: internal/satellite/module/processor/processor.go [hunk 3]
```diff
@@ -83,3 +84,18 @@ func (p *Processor) Boot(ctx context.Context) {
...
+
+func (p *Processor) syncProcessor(e *v1.SniffData) (*v1.SniffData, error) {
+	// build contest
+	c := &event.OutputEventContext{
+		Offset:  "",
+		Context: make(map[string]*v1.SniffData),
+	}
+	c.Put(e)
+	// processing the event with filters, that put the necessary events to OutputEventContext.
+	for _, f := range p.runningFilters {
```
Comment (EvanLjp) [inline internal/satellite/module/processor/processor.go:96]: sync process means invoke remote method, so the lines[89-99] should be removed

File: internal/satellite/module/sender/sender.go [hunk 4]
```diff
@@ -216,3 +217,12 @@ func (s *Sender) consume(batch *buffer.BatchBuffer) {
 func (s *Sender) InputDataChannel() chan<- *event.OutputEventContext {
 	return s.input
 }
+
+func (s *Sender) SyncProcess(data *v1.SniffData) (*v1.SniffData, error) {
```
Comment (EvanLjp) [inline internal/satellite/module/sender/sender.go:221]: How about adding a new interface in forwarder API called supportedSyncInvoke to select the sync forwarder

File: internal/satellite/module/api/module.go [hunk 5]
```diff
@@ -27,7 +27,7 @@ var ShutdownHookTime = time.Second * 5
 
 // Module id a custom plugin interface, which defines the processing.
 type Module interface {
-
+	DependencyInjection(modules ...Module)
```
Comment (wu-sheng) [inline internal/satellite/module/api/module.go:30]: Miss comments for core level interface.

Comment (wu-sheng) [reply internal/satellite/module/api/module.go:30]: If you want to run DI, how could you figure out the elements of different indices, and do the typecast?

File: internal/satellite/module/gatherer/fetcher_gatherer.go [hunk 6]
```diff
@@ -117,3 +121,7 @@ func (f *FetcherGatherer) OutputDataChannel() <-chan *queue.SequenceEvent {
 func (f *FetcherGatherer) Ack(lastOffset event.Offset) {
 	f.runningQueue.Ack(lastOffset)
 }
+
+func (f *FetcherGatherer) DependencyInjection(modules ...module.Module) {
+	f.processor = modules[0].(processor.Processor)
```
Comment (wu-sheng) [inline internal/satellite/module/gatherer/fetcher_gatherer.go:126]: I can see a `[0]` directly, could you explain the DI design logic?

Comment (EvanLjp) [reply internal/satellite/module/gatherer/fetcher_gatherer.go]: please recheck codes.

**Comment by wu-sheng** [comment]
@EvanLjp I am still confused about `DependencyInjection` method. Why do you think we need this kind of abstract way to inject dependencies? Rather than the previous ways.

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
@EvanLjp Thanks for making clean APIs back.

**Pull request merged**